### ISSUE TITLE
Can I use only two GPUs to train this network?

### ISSUE THREAD

**Issue by huangerjun** [yysijie/st-gcn#44]
Can I use only two GPUs to train this network?

**Comment by dracarys983** [comment]
**TL;DR** I believe you can run it on two Titan X or two 1080 Ti GPUs, with a batch size of 32. Or sample frames from each video to reduce input length of videos.

The input contains entire spatio-temporal information (entire videos) with the length of each video being a fixed number (padded to make the lengths same for batching). Hence, reducing the batch size will result in less GPU memory being used. Assuming the availability of two GPUs with 11 or 12 GB memory, this is the easiest way to go.
For GPUs with smaller memory config, you have to reduce the batch size accordingly. However, note that a smaller batch size will result in the loss requiring more time to reach stability. It will also have more fluctuations while training compared to a larger batch size.

An alternative way (I am not sure about the performance) would be to randomly sample frames from the videos and reduce the input length of videos. For this, you can divide all the videos into equal number of chunks and sample two or three frames randomly from each chunk. In this way, you can produce more data (augmentation) as well as reduce the input length of videos in order to still train it with a batch size more than 32 using only two GPUs.

**Comment by yysijie** [comment]
@huangerjun,
Thank you for your interest.
Sure, please use smaller batch size for training and test, like this:
`python main.py recognition -c config/st_gcn/kinetics-skeleton/train.yaml --device 0 1 --batch_size 32 --test_batch_size 32`

**Comment by yysijie** [comment]
@dracarys983 
Yes, the first solution is OK. We never carefully adjust hyper-parameters. Maybe you can find a better setting.
